vm/src/stdlib/mod.rs

```diff
@@ -71,108 +77,116 @@ mod winreg;
 
 pub type StdlibInitFunc = Box<py_dyn_fn!(dyn Fn(&VirtualMachine) -> PyObjectRef)>;
 
-pub fn get_module_inits() -> HashMap<String, StdlibInitFunc, ahash::RandomState> {
-    #[allow(unused_mut)]
-    let mut modules = hashmap! {
-        hasher = ahash::RandomState::default(),
-        "array".to_owned() => Box::new(array::make_module) as StdlibInitFunc,
-        "atexit".to_owned() => Box::new(atexit::make_module),
-        "binascii".to_owned() => Box::new(binascii::make_module),
-        "_collections".to_owned() => Box::new(collections::make_module),
-        "_csv".to_owned() => Box::new(csv::make_module),
-        "dis".to_owned() => Box::new(dis::make_module),
-        "errno".to_owned() => Box::new(errno::make_module),
-        "_functools".to_owned() => Box::new(functools::make_module),
-        "hashlib".to_owned() => Box::new(hashlib::make_module),
-        "itertools".to_owned() => Box::new(itertools::make_module),
-        "_io".to_owned() => Box::new(io::make_module),
-        "_json".to_owned() => Box::new(json::make_module),
-        "marshal".to_owned() => Box::new(marshal::make_module),
-        "math".to_owned() => Box::new(math::make_module),
-        "_operator".to_owned() => Box::new(operator::make_module),
-        "_platform".to_owned() => Box::new(platform::make_module),
-        "regex_crate".to_owned() => Box::new(re::make_module),
-        "_random".to_owned() => Box::new(random::make_module),
-        "_serde_json".to_owned() => Box::new(serde_json::make_module),
-        "_sre".to_owned() => Box::new(sre::make_module),
-        "_string".to_owned() => Box::new(string::make_module),
-        "_struct".to_owned() => Box::new(pystruct::make_module),
-        "time".to_owned() => Box::new(time_module::make_module),
-        "_weakref".to_owned() => Box::new(weakref::make_module),
-        "_imp".to_owned() => Box::new(imp::make_module),
-        "unicodedata".to_owned() => Box::new(unicodedata::make_module),
-        "_warnings".to_owned() => Box::new(warnings::make_module),
-        "zlib".to_owned() => Box::new(zlib::make_module),
-        crate::sysmodule::sysconfigdata_name() => Box::new(sysconfigdata::make_module),
-    };
-
-    // Insert parser related modules:
-    #[cfg(feature = "rustpython-ast")]
-    {
-        modules.insert(
-            "_ast".to_owned(),
-            Box::new(ast::make_module) as StdlibInitFunc,
-        );
-    }
-
-    #[cfg(feature = "rustpython-parser")]
-    {
-        modules.insert("keyword".to_owned(), Box::new(keyword::make_module));
-    }
+pub type StdlibMap = HashMap<Cow<'static, str>, StdlibInitFunc, ahash::RandomState>;
```

Comment:
what's the benefit to have this as `Cow`? Does this not fit for `&'static str` but still Cow is better than String?

Reply:
Yeah, cause we still want people to be able to input dynamic builtin modules (e.g. `_sysconfigdata_{abi}_{os}_{triple}`) but there's really no need to allocate a String for every key since *most* of them will be static.

Reply:
A bit doubtful for the benefit. The number of modules will not explode to big numbers. So it will be less than a few kb, but Cow might have performance drop for every access. (I have no idea for that. Is it free for accessing?)
If it does have benefit, `abi`, `os` and `triple` are const, so `_sysconfigdata_{abi}_{os}_{triple}` looks like to possibly be a `&'str` (const_concat: https://github.com/Vurich/const-concat). Though users must make every module name `&'static str` in this case, it will not be a big deal with leaking.

Reply:
`Cow<str>` is slightly bigger in size - 32 bytes vs String is 24 bytes - but that's offset by not needing to allocate strings separately. And access is pretty much free, and it wouldn't be hot anyway, since after stdlib_inits is consulted once the module is put into sys.modules.

Reply:
We could also use `beef::Cow` which implements a `Cow<str>` in only 24 bytes.